### PR TITLE
feat(IamPolicyChecks): implement Telemetry, update error colors, fix problems line number

### PR DESCRIPTION
**Pull request by kevluu-aws**
Telemetry implementation, updating response colors to be shown in the Webview, and return intended line numbers in problems panel.

Corresponding PR for new metrics in `aws-toolkit-common`: https://github.com/aws/aws-toolkit-common/pull/735
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (hayemaxi) [inline packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:84]: instead of using `.emit()`, use `.record()`. This will handle errors automatically. There are many examples in the codebase and it is also explained in `telemetry.md`.

Applicable throughout PR.

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (hayemaxi) [inline packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:121]: Type your elementIds.

Example:
https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/login/webview/vue/types.ts#L48

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (hayemaxi) [inline packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:149]: Can you maybe check for instances of `ExpiredTokenException` or similar?

File: packages/core/src/shared/telemetry/vscodeTelemetry.json
Comment (hayemaxi) [inline packages/core/src/shared/telemetry/vscodeTelemetry.json:1288]: The `accessanalyzer` prefixes for the metadata are unnecessary

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (kevluu-aws) [reply packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:84]: One of the reasons I strayed away from `.record()` is because the tool we use returns an "error" when we discover findings, under a different exit code. We check for the exit code to handle it gracefully, but if I used `.run()` and `.record()` there is no way for me to specify that it actually `Succeeded`.
I found that I am unable to set the `result` or `reason` when using `.record()`

Comment (kevluu-aws) [reply packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts]: Using `.record()` now

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (kevluu-aws) [reply packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts]: Yeah, that should do that trick for this instance. Unfortunately, I will have to parse it for the CLI output as we get it back as a blob.

Comment (kevluu-aws) [reply packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts]: This does the trick for this instance, though I will still need to parse for the message for the CLI errors below, since they only give back string blobs.

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (hayemaxi) [inline packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:435]: nit: you can add this to your first record on line 415. Fields can be re-recorded as much as you want, and the latest recording will be the one that makes it into the metric.

File: packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts
Comment (hayemaxi) [inline packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts:672]: nit: same comment as before, try to prefer comparing to an exception type if possible rather than the message.

**Comment by hayemaxi** [comment]
Lgtm, waiting for an approval from your team.

**Review by hayemaxi — APPROVED** [review]

**Review by mponaws — APPROVED** [review]

File: packages/core/src/shared/telemetry/vscodeTelemetry.json
Comment (dhruvigajjar) [inline packages/core/src/shared/telemetry/vscodeTelemetry.json:6]: can you try and put all metrics in common only. makes it easier to keep track

Comment (kevluu-aws) [reply packages/core/src/shared/telemetry/vscodeTelemetry.json:6]: I created these to have the metrics available in my dev environment. I am still waiting for the common [changes](https://github.com/aws/aws-toolkit-common/pull/735) to be merged, which has the same metrics.

**Pull request merged**
